src/compiler/checker.ts

```diff
@@ -7302,7 +7302,11 @@ namespace ts {
                 if (flags & TypeFlags.Null) typeSet.containsNull = true;
                 if (!(flags & TypeFlags.ContainsWideningType)) typeSet.containsNonWideningType = true;
             }
-            else if (!(flags & TypeFlags.Never)) {
+            else if (!(flags & TypeFlags.Never || flags & TypeFlags.Intersection && every((<IntersectionType>type).types, isUnitType))) {

```

Comment:
Rather than checking if `every` type is unit, is it not sufficient to check if any two members are unit? `number & 3 & 4` is as vacuous as `3 & 4`, right?